### ISSUE TITLE
Avoid conflicts with other patch applier plugins

### ISSUE THREAD

**Issue by allanpaiste** [vaimo/composer-patches#66]
**Is your feature request related to a problem? Please describe.**

The plugin clashes with other Composer plugins that deal with similar topic - making it unviable to use the module as dependency. 

The issue comes from using general extra.patches for the patches reference.

**Describe the solution you'd like**

Add configuration option on package-level where the plugin would only deal with applying patches from the packages that use it as dependency. When the plugin is not required by project root and all of the packages claim exclusive usage of the patcher, avoid applying patches from anywhere else.

In short, this would mean that the patch applier would only work against modules that directly, explicitly require it.

Additionally: output a warning when other plugins with same purpose present in a project. In that case allow the more focused functionality to be enabled via ROOT package configuration.

**Describe alternatives you've considered**

Another alternative would be just to warn the user and not provide any way to continue which would just be more inconvenient. 

Yet another would be to switch the configuration keys for patch definition to completely new keys.
